python/ray/node.py

```diff
@@ -137,6 +139,28 @@ def __init__(self,
         assert self.max_bytes >= 0
         assert self.backup_count >= 0
 
+        if head:
+            ray_params.update_if_absent(num_redis_shards=1)
+            gcs_server_port = os.getenv(
+                ray_constants.GCS_PORT_ENVIRONMENT_VARIABLE)
+            if gcs_server_port:
+                ray_params.update_if_absent(gcs_server_port=gcs_server_port)
+            if use_gcs_for_bootstrap():
```

Comment:
Is this block from the prototype? Probably doesn't make sense to force override `ray_params.gcs_server_port` with `ray_constants.GCS_DEFAULT_PORT`.

Also this function seems to adopt the style of conditional initialization for each param, instead of a long `if head: ... else ...` block. I think it may improve readability to initialize `self._webui_url` and port in separate if blocks.